### ISSUE TITLE
Difficulty installing : cuDNN launch failure

### ISSUE THREAD

**Issue by b2jia** [tensorflow/gnn#480]
Really intrigued by the `encoder-process-decoder` architecture and I'm really interested in trying out this model!

I know these "installation" issues are highly user dependent, extremely variable, and clutters the issue tab. But I'm having such a hard time getting started amid version conflicts and installation errors. Can the authors suggest a "recipe" of compatible libraries (ie. Tensorflow version, CUDA) considering `tensorflow_gnn`? For what it's worth, my models written in `tensorflow` seem to work fine.

- Python 3.9.16
- CUDA 11.4
- tensorflow 2.12.0
- tensorflow-gnn 0.5.0

is what I'm currently using, but keeps getting stuck at this error when running the `shortest_path` demo:
```
---------------------------------------------------------------------------
InternalError                             Traceback (most recent call last)
Cell In[7], line 31
     29 # Call the model once to inspect all of the variables.
     30 task_graph = next(iter(task_dataset))
---> 31 trainable_gnn(build_initial_hidden_state(task_graph))
     32 for var in trainable_gnn.variables:
     33   print(var.name, var.shape, var.dtype)

File ~/anaconda3/envs/tensorflowgnn/lib/python3.9/site-packages/keras/utils/traceback_utils.py:70, in filter_traceback.<locals>.error_handler(*args, **kwargs)
     67     filtered_tb = _process_traceback_frames(e.__traceback__)
     68     # To get the full stack trace, call:
     69     # `tf.debugging.disable_traceback_filtering()`
---> 70     raise e.with_traceback(filtered_tb) from None
     71 finally:
     72     del filtered_tb

Cell In[2], line 248, in EncodeProcessDecode.call(self, input_graph)
    247 def call(self, input_graph: tfgnn.GraphTensor) -> tfgnn.GraphTensor:
--> 248   latent_graph = self._encoder(input_graph)
    249   for processor in self._processors:
    250     residual_graph = processor(latent_graph)

File ~/anaconda3/envs/tensorflowgnn/lib/python3.9/site-packages/tensorflow_gnn/keras/layers/map_features.py:216, in MapFeatures.call(self, graph)
    214 context_features = None
    215 if self._context_model is not None:
--> 216   context_features = _call_model(self._context_model, graph.context,
    217                                  logging_name="context")
    219 node_set_features = {}
    220 for node_set_name, node_set in graph.node_sets.items():

File ~/anaconda3/envs/tensorflowgnn/lib/python3.9/site-packages/tensorflow_gnn/keras/layers/map_features.py:285, in _call_model(model, graph_piece, logging_name)
    279 if expected_features != actual_features:
    280   raise ValueError(
    281       f"The feature set of {logging_name} has changed since this layer "
    282       f"was built. Expected {sorted(expected_features)}, "
    283       f"got {sorted(actual_features)}.")
--> 285 return model(graph_piece)

InternalError: Exception encountered when calling layer 'encode_process_decode/encoder/context/layer_norm' (type LayerNormalization).

{{function_node __wrapped__FusedBatchNormV3_device_/job:localhost/replica:0/task:0/device:GPU:0}} cuDNN launch failure : input shape ([1,1,128,1]) [Op:FusedBatchNormV3]

Call arguments received by layer 'encode_process_decode/encoder/context/layer_norm' (type LayerNormalization):
  • inputs=tf.Tensor(shape=(1, 128), dtype=float32)
```

**Comment by arnoegw** [comment]
Hi @b2jia, thanks for reaching out!

I take it that code is from [graph_network_shortest_path.ipynb](https://colab.research.google.com/github/tensorflow/gnn/blob/master/examples/notebooks/graph_network_shortest_path.ipynb). I got it to run with GPU on Google Colab (as linked) after replacing the `pip install`line by

```
pip install protobuf~=3.20.0 grpcio-tools~=1.43.0 apache-beam~=2.46.0 tensorflow-gnn~=0.5.0
```

The resulting versions are

  * Python 3.10.12, reported by `import sys; print(sys.version)`
  * CUDA 11.8, reported by `!nvcc --version`
  * TensorFlow 2.12.0; reported by `import tensorflow as tf; print(tf.__version__)`
  * TF-GNN 0.5.0

Does that work for you?

PS: Some separate cleanup is needed to make `pip install tensorflow-gnn` work directly without specifying the other package versions.

**Comment by arnoegw** [comment]
@mihirparadkar  just released pip package `tensorflow-gnn==0.5.1`, and [graph_network_shortest_path.ipynb](https://colab.research.google.com/github/tensorflow/gnn/blob/master/examples/notebooks/graph_network_shortest_path.ipynb) should work out of the box again.

**Comment by b2jia** [comment]
Thank you for your support on this!
